### ISSUE TITLE
Monorepo build failing after version 15.0.3 onwards

### ISSUE THREAD

**Issue by digoburigo** [vercel/next.js#73578]
### Link to the code that reproduces this issue

https://github.com/t3-oss/create-t3-turbo

### To Reproduce

1. Clone repo and install the deps with `pnpm`
2. Update to version `15.0.2`, build it and it works
OBS: change the root `package.json` to include the build command  to only build nextjs
`"build:next": "CI=true turbo run build -F @acme/nextjs...",`
3. Update to version `15.0.3` or `15.0.4`, build it again and you will see that the external `packages` are not found and the build fails, for example:
```bash
@acme/nextjs:build:    Creating an optimized production build ...
@acme/nextjs:build: Failed to compile.
@acme/nextjs:build:
@acme/nextjs:build: ./src/app/layout.tsx
@acme/nextjs:build: Module not found: Can't resolve '@acme/ui/theme'
@acme/nextjs:build:
@acme/nextjs:build: https://nextjs.org/docs/messages/module-not-found
@acme/nextjs:build:
@acme/nextjs:build: ./src/app/layout.tsx
@acme/nextjs:build: Module not found: Can't resolve '@acme/ui/toast'
@acme/nextjs:build:
@acme/nextjs:build: https://nextjs.org/docs/messages/module-not-found
```

### Current vs. Expected behavior

I expected to build it normally like in the `15.0.2` version

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.1.0: Thu Oct 10 21:03:15 PDT 2024; root:xnu-11215.41.3~2/RELEASE_ARM64_T6000
  Available memory (MB): 16384
  Available CPU cores: 8
Binaries:
  Node: 22.12.0
  npm: 10.9.2
  Yarn: N/A
  pnpm: 9.14.4
Relevant Packages:
  next: 15.0.4 // Latest available version is detected (15.0.4).
  eslint-config-next: N/A
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.6.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Module Resolution, Output (export/standalone), Webpack

### Which stage(s) are affected? (Select all that apply)

next build (local)

**Comment by mischnic** [comment]
(This is not a Turbopack but a Webpack build, you're just using Turbo as a task runner)

This is effectively a duplicate of https://github.com/vercel/next.js/issues/71001, just for Webpack instead of Turbopack

**Comment by Christophevvv** [comment]
I had the same issue.
What I did to get unblocked for now is change the exports in packages/ui/package.json
```
 "./*": {
      "types": "./dist/src/*.d.ts",
      "default": [
        "./src/*.ts",
        "./src/*.tsx"
      ]
    }
```
The issue seems to be with the array where only the first item will be used, so if you put the ./src/*.tsx one before the ./src/*.ts imports like @acme/ui/toast etc will resolve fine. If you want to import a .ts file an extra workaround is required.

**Comment by samcx** [comment]
@digoburigo Marking this as a duplicate of a newer (only 1 day newer) issue since that one has more reactions.
